elyra-ai/elyra
# Fix debugger button state on kernel selection change

Fixes #2916
Changing kernel selection now updates the debug button state accordingly.
![debug-button](https://user-images.githubusercontent.com/25207344/197063678-7d1910bc-597d-4713-a556-18682f076f89.gif)

### What changes were proposed in this pull request?
Updated  the `updateDebugger`  method logic to handle a few areas where UI update was not being triggered for the expected workflow.

### How was this pull request tested?
Manually tested by mocking a python kernel without debug support as below:
1. jupyter kernelspec list
2. copy default python3 kernel directory and content to the same kernels/ directory
3. Change kernel.json content to:
```
 "display_name": "Python 3 - test",
 "language": "python",
 "metadata": {
  "debugger": false
 }
```
When changing kernels verify the debug button state (enabled/disabled) gets updated accordingly (sometimes it takes a few ms to update).
 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.
